packages/headless/src/features/product-listing/product-listing-analytics.ts

```diff
@@ -18,15 +19,89 @@ export const logProductListing = (): ProductListingAction =>
     (getState) => new ProductListingAnalyticsProvider(getState)
   );
 
-export const logDocumentOpen = (result: Result): ClickAction =>
+export const logDocumentOpen = (
+  productRec: ProductRecommendation
+): ClickAction =>
   makeAnalyticsAction(
     'analytics/productListing/open',
     AnalyticsType.Click,
     (client, state) => {
-      validateResultPayload(result);
+      validateProductRecommendationPayload(productRec);
       return client.makeDocumentOpen(
-        partialDocumentInformation(result, state),
-        documentIdentifier(result)
+        partialProductRecommendationInformation(productRec, state),
+        {
+          contentIDKey: 'permanentid',
+          contentIDValue: productRec.permanentid,
+        }
       );
     }
   );
+
+export const partialProductRecommendationInformation = (
+  productRec: ProductRecommendation,
+  state: Partial<ProductListingAppState>
+): PartialDocumentInformation => {
+  const paginationBasedIndex = (index: number) =>
+    index + (state.pagination?.firstResult ?? 0);
+
+  let productRecIndex = -1;
+
+  const parentResults = state.productListing?.products;
+  productRecIndex = findPositionWithPermanentid(productRec, parentResults);
+
+  if (productRecIndex < 0) {
+    productRecIndex = findPositionInChildResults(productRec, parentResults);
+  }
+
+  if (productRecIndex < 0) {
+    // ¯\_(ツ)_/¯
+    productRecIndex = 0;
+  }
+
+  return buildPartialProductRecommendationInformation(
+    productRec,
+    paginationBasedIndex(productRecIndex)
+  );
+};
+
+function findPositionWithPermanentid(
+  targetProductRec: ProductRecommendation,
+  productRecs: ProductRecommendation[] = []
+) {
+  return productRecs.findIndex(
+    ({permanentid}) => permanentid === targetProductRec.permanentid
+  );
+}
+
+function findPositionInChildResults(
+  targetProductRec: ProductRecommendation,
+  parentResults: ProductRecommendation[] = []
+) {
+  for (const [i, parent] of parentResults.entries()) {
+    const children = parent.childResults;
+    const childIndex = findPositionWithPermanentid(targetProductRec, children);
+    if (childIndex !== -1) {
+      return i;
+    }
+  }
+
+  return -1;
+}
+
+function buildPartialProductRecommendationInformation(
+  productRec: ProductRecommendation,
+  resultIndex: number
+): PartialDocumentInformation {
+  return {
+    collectionName: '',
+    documentAuthor: '',
+    documentPosition: resultIndex + 1,
+    documentTitle: productRec.ec_name || '',
+    documentUri: productRec.documentUri,
+    documentUriHash: productRec.documentUriHash,
+    documentUrl: productRec.clickUri,
+    rankingModifier: '',
+    sourceName: '',
+    queryPipeline: getPipelineInitialState(),
+  };
```

Comment:
I wanted to reuse the `PartialDocumentInformation` but it leaves some blank properties. I'm open to either create a new interface for product recs or trying to make PartialDocumentInformation more generic.

Reply:
Either of those would be breaking changes I reckon. If my making `PartialDocumentInformation` more generic you mean removing properties from the interface.

At face value, I think the proper approach would be to create a new interface specifically for product recs. But this would have to be done in a major version, and tracked here: https://coveord.atlassian.net/browse/KIT-2157

As for now, I think the only thing that can be done in a backward compatible manner is to do what you've done here.

Reply:
I think creating a new interface would be a good idea.

Reply:
The new interface would be in coveo.analytics and will require bigger changes. I think I'd go with Oli for now and keep it as is.